tests/softreboot_test.go

```diff
@@ -67,91 +78,77 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 
 		var vmi *v1.VirtualMachineInstance
 
-		runVMI := func(withGuestAgent bool, ACPIEnabled bool) {
-			if withGuestAgent {
-				vmi = tests.NewRandomFedoraVMIWithGuestAgent()
-			} else {
-				vmi = tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskCirros))
-				tests.AddUserData(vmi, "cloud-init", "#!/bin/bash\necho 'hello'\n")
-			}
-			if !ACPIEnabled {
-				vmi.Spec.Domain.Features = &v1.Features{
-					ACPI: v1.FeatureState{Enabled: &ACPIEnabled},
-				}
-			}
-			tests.RunVMIAndExpectLaunch(vmi, 360)
-		}
-
 		When("soft reboot vmi with agent connected via API", func() {
 			It("should succeed", func() {
-				runVMI(true, false)
+				vmi = tests.RunVMIAndExpectLaunch(libvmi.NewTestToolingFedora(withoutACPI()), vmiLaunchTimeout)
```

Comment:
Nit: `NewFedora()` just calls `NewTestToolingFedora()` and could maybe improve readability

Reply:
oh, I did not notice this. I expected `NewFedora` to give me a vanilla Fedora; the test here explicitly expects the guest agent.

I now see that commit 36aba3ec0d eliminated the distinction between the two; but kept the two functions. I see suspect there is a case for total elimination of `NewTestToolingFedora`, however I would keep it for a separate PR - I would like to consider if we do not loose information by this.